### PR TITLE
Remove `default.qubit.torch`

### PR DESCRIPTION
**Pull request by albi3ro**
Removes `default.qubit.torch`.  Follows on from the removal of `default.qubit.tf`.  Each interface device will be its own PR to keep the scope of the changes managable.

[sc-72784]

### CODE REVIEW

**Review by astralcai — APPROVED** [review]
🎉

File: tests/templates/test_subroutines/test_reflection.py [hunk 1]
```diff
@@ -164,11 +164,10 @@ def test_lightning_qubit(self):
 
     @pytest.mark.autograd
     @pytest.mark.parametrize("shots", [None, 50000])
-    @pytest.mark.parametrize("device", ["default.qubit", "default.qubit.legacy"])
```
Comment (PietropaoloFrisoni) [inline tests/templates/test_subroutines/test_reflection.py:167]: Is there a specific reason why this was also tested with `default.qubit.legacy`?

**Review by PietropaoloFrisoni — APPROVED** [review]
Nice!

File: tests/templates/test_subroutines/test_reflection.py (hunk 1 above)
Comment (albi3ro) [reply tests/templates/test_subroutines/test_reflection.py:167]: probably just sanity checks to make sure nothign wonky is going on.

**Pull request merged**